pkg/kubectl/resource/builder.go

```diff
@@ -656,6 +671,9 @@ func (b *Builder) visitorResult() *Result {
 // inputs are consumed by the first execution - use Infos() or Object() on the Result to capture a list
 // for further iteration.
 func (b *Builder) Do() *Result {
+	for ix := range b.paths {
```

Comment:
This is more of a stylistic issue, but I'd prefer transforms to be passed into the constructors of the Path visitors directly, rather than set lazily later on.  Path visitors taking a transform as an optional argument to their constructor isn't unreasonable, which removes the need for the setter here.